internal/subscription/database/subscription_repository.go

```diff
@@ -114,12 +114,12 @@ func (r SubscriptionRepository) UpdateSubscription(sub *domain.Subscription) err
 		sub.ExpiresAt,
 	}
 	if err := stmt.Get(sub, args...); err != nil {
-		return errors.WrapErrorf(err, errors.ErrCodeUnknown, "error updating subscription")
+		return domain.Subscription{}, errors.WrapErrorf(err, errors.ErrCodeUnknown, "error updating subscription")
 	}
-	return nil
+	return r.Subscription(sub.UUID)
```

Comment:
Is this a get? But why? The update query isn't already returning the updated object?

Reply:
We can't return course_uuid and matrix_uuid (relationship), so I called the getSubscription here. WDYT?

Reply:
I see... 🤔  I guess that we can't update the query to return the inner join as well, right?

Reply:
@arianerocha WDYT?

Reply:
I tried to create a query to update and get all the values (including user_uuid, course_uuid and matrix_uuid), but the query was complicated, so I found it easier to use the `Matrix(matrix_uuid uuid.UUID)` function.

@arianerocha WDYT?

Reply:
I didn't want to say it depends, but it depends. I think it would be simpler to return just the updated object, unless in specific cases you really need to return the object including the relationships, but i'm not sure if I could do this in the update just with return value,
You may have to add additional treatment, I would go for the simply version

Reply:
Yeah, I see your point. I think that there is no harm in using the `r.Subscription` to get the value with the relationships for now. Getting just the uuids seems a little bit useless IMHO. But I don't know any good reasons to defend my opinion, it is just a guess... If we feel that it doesn't make sense we can remove it.

Reply:
https://github.com/sumelms/microservice-course/pull/28/commits/0e1e16db83709e0878f3b4bbf9e73085d95a5b8b